packages/test/test-utils/src/timeoutUtils.ts

```diff
@@ -67,11 +62,11 @@ class TestTimeout {
 		// Set up timer to reject near the test timeout.
 		this.timer = setTimeout(() => {
 			this.deferred.reject(this);
-			this.rejected = true;
 		}, this.timeout);
 	}
 	private clearTimer() {
 		if (this.timer) {
+			this.deferred = new Deferred();
```

Comment:
Again probably fine functionally (I think this is creating a new Deferred per test given how Mocha is calling into these functions) but I suspect probably defeats the original purpose of this class sharing a static instance.

Reply:
the static instance is more about binding to the timeout of the test's timeout, and timer management, as this hooks into mocha, reusing the promise isn't important as far as i can tell.